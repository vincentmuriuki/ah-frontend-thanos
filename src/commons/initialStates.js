const initialState = {
  articleReducer: {
    article: {
      id: 0,
      tag_list: [],
      slug: '',
      title: '',
      description: '',
      body: '',
      created_at: '',
      updated_at: '',
      image_url: '',
      audio_url: '',
      read_time: '',
      likes: 0,
      dislikes: 0,
      likeDislikeStatus: '',
      views_count: '',
      rating: 0,
      author: {
        id: 0,
        username: '',
        email: '',
      },
    },
    articles: [],
    errorMessage: '',
  },
  deleteArticleReducer: {},
  socialLoginReducer: { isLoggedIn: false },
  loginReducer: { errorMessage: '', successMessage: '', user_details: '' },
  userReducer: {
    freshUser: { email: '', password: '', username: '' },
  },
};

export const sampleArticle = {
  id: 1,
  tag_list: ['tag'],
  slug: 'title-sample',
  title: 'title sample',
  description: 'desc',
  body: 'body',
  created_at: '12-12-2018',
  updated_at: '12-12-2018',
  image_url: 'http://test.png',
  audio_url: 'http://test.mp3',
  read_time: '0 min',
  likes: 12,
  dislikes: 2,
  likeDislikeStatus: 'dislike',
  views_count: 0,
  rating: 0,
  author: {
    id: 2,
    username: 'janedoe',
    email: 'janedoe@gmail.com',
  },
};

export const sampleArticle2 = {
  ...sampleArticle,
  likeDislikeStatus: 'like',
};

export const initialStateWithSample = {
  article: sampleArticle,
};

export const initialStateWithSample2 = {
  article: sampleArticle2,
};

export default initialState;
