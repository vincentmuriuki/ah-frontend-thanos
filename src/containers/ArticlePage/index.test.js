import React from 'react';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import ArticlePageConnected, { ArticlePage, mapDispatchToProps } from '.';
import { sampleArticle } from '../../commons/initialStates';

const testClickButton = (wrapper, buttonId, mockMethod) => {
  wrapper.find(buttonId).simulate('click');
  expect(mockMethod).toHaveBeenCalled();
};

describe('<ArticlePage />', () => {
  let props;
  let initialState;
  let wrapper;
  let mockStore;
  let likeDislikeArticleDispatch;

  beforeEach(() => {
    props = {
      match: { params: { articleId: sampleArticle.id } },
      likeDislikeArticleDispatch: jest.fn(),
      getArticleDispatch: jest.fn(),
      getLikeDislikeStatusDispatch: jest.fn(),
      article: sampleArticle,
    };
    initialState = { articleReducer: { article: sampleArticle }, deleteArticleReducer: {} };
    mockStore = configureStore([thunk]);

    likeDislikeArticleDispatch = jest.fn();
    const {
      match, getArticleDispatch, getLikeDislikeStatusDispatch, article,
    } = props;
    wrapper = mount(
      <Provider store={mockStore(initialState)}>
        <ArticlePage
          match={match}
          likeDislikeArticleDispatch={likeDislikeArticleDispatch}
          getArticleDispatch={getArticleDispatch}
          getLikeDislikeStatusDispatch={getLikeDislikeStatusDispatch}
          article={article}
        />
      </Provider>,
    );
  });

  it('renders the container', () => {
    const { match } = props;
    wrapper = mount(
      <Provider store={mockStore(initialState)}>
        <ArticlePageConnected match={match} />
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should trigger handleLike method', () => {
    testClickButton(wrapper, 'button#bt-like', likeDislikeArticleDispatch);
  });

  it('should trigger handleDisLike method', () => {
    testClickButton(wrapper, 'button#bt-dislike', likeDislikeArticleDispatch);
  });

  it('should check for the presence of like/dislike dispatch', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).likeDislikeArticleDispatch({});
    expect(dispatch.mock.calls[0][0]).toBeDefined();
  });
});
