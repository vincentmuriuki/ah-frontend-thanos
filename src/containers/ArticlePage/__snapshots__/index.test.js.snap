// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticlePage /> renders the container 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(ArticlePage)
    match={
      Object {
        "params": Object {
          "articleId": 1,
        },
      }
    }
  >
    <ArticlePage
      article={
        Object {
          "audio_url": "http://test.mp3",
          "author": Object {
            "email": "janedoe@gmail.com",
            "id": 2,
            "username": "janedoe",
          },
          "body": "body",
          "created_at": "12-12-2018",
          "description": "desc",
          "dislikes": 2,
          "id": 1,
          "image_url": "http://test.png",
          "likeDislikeStatus": "dislike",
          "likes": 12,
          "rating": 0,
          "read_time": "0 min",
          "slug": "title-sample",
          "tag_list": Array [
            "tag",
          ],
          "title": "title sample",
          "updated_at": "12-12-2018",
          "views_count": 0,
        }
      }
      getArticleDispatch={[Function]}
      getLikeDislikeStatusDispatch={[Function]}
      likeDislikeArticleDispatch={[Function]}
      match={
        Object {
          "params": Object {
            "articleId": 1,
          },
        }
      }
    >
      <Article
        article={
          Object {
            "audio_url": "http://test.mp3",
            "author": Object {
              "email": "janedoe@gmail.com",
              "id": 2,
              "username": "janedoe",
            },
            "body": "body",
            "created_at": "12-12-2018",
            "description": "desc",
            "dislikes": 2,
            "id": 1,
            "image_url": "http://test.png",
            "likeDislikeStatus": "dislike",
            "likes": 12,
            "rating": 0,
            "read_time": "0 min",
            "slug": "title-sample",
            "tag_list": Array [
              "tag",
            ],
            "title": "title sample",
            "updated_at": "12-12-2018",
            "views_count": 0,
          }
        }
        onLikeDislike={[Function]}
      >
        <div
          className="full-article"
        >
          <div
            className="article-title"
          >
            <h1>
              title sample
            </h1>
            <p
              className="text-white"
            >
              desc
            </p>
          </div>
          <div
            className="container"
          >
            <div
              className="article-image text-center"
            >
              <img
                alt="poster"
                src="http://test.png"
              />
            </div>
            <div
              className="article-small-details"
            >
              <span
                key="author"
              >
                <FontAwesomeIcon
                  border={false}
                  className=""
                  fixedWidth={false}
                  flip={null}
                  icon="user"
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-user fa-w-14 "
                    data-icon="user"
                    data-prefix="fas"
                    role="img"
                    style={Object {}}
                    viewBox="0 0 448 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                </FontAwesomeIcon>
                <div
                  className="author"
                >
                  janedoe
                </div>
              </span>
              <span
                key="readtime"
              >
                <FontAwesomeIcon
                  border={false}
                  className=""
                  fixedWidth={false}
                  flip={null}
                  icon="clock"
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-clock fa-w-16 "
                    data-icon="clock"
                    data-prefix="fas"
                    role="img"
                    style={Object {}}
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm57.1 350.1L224.9 294c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h48c6.6 0 12 5.4 12 12v137.7l63.5 46.2c5.4 3.9 6.5 11.4 2.6 16.8l-28.2 38.8c-3.9 5.3-11.4 6.5-16.8 2.6z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                </FontAwesomeIcon>
                <div
                  className="readtime"
                >
                  0 min min read
                </div>
              </span>
              <span
                key="created-at"
              >
                <FontAwesomeIcon
                  border={false}
                  className=""
                  fixedWidth={false}
                  flip={null}
                  icon="calendar"
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-calendar fa-w-14 "
                    data-icon="calendar"
                    data-prefix="fas"
                    role="img"
                    style={Object {}}
                    viewBox="0 0 448 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12 192h424c6.6 0 12 5.4 12 12v260c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V204c0-6.6 5.4-12 12-12zm436-44v-36c0-26.5-21.5-48-48-48h-48V12c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v52H160V12c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v52H48C21.5 64 0 85.5 0 112v36c0 6.6 5.4 12 12 12h424c6.6 0 12-5.4 12-12z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                </FontAwesomeIcon>
                <div
                  className="created-at"
                >
                  Created: 12-12-2018
                </div>
              </span>
              <span
                key="updated-at"
              >
                <FontAwesomeIcon
                  border={false}
                  className=""
                  fixedWidth={false}
                  flip={null}
                  icon="calendar"
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-calendar fa-w-14 "
                    data-icon="calendar"
                    data-prefix="fas"
                    role="img"
                    style={Object {}}
                    viewBox="0 0 448 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12 192h424c6.6 0 12 5.4 12 12v260c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V204c0-6.6 5.4-12 12-12zm436-44v-36c0-26.5-21.5-48-48-48h-48V12c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v52H160V12c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v52H48C21.5 64 0 85.5 0 112v36c0 6.6 5.4 12 12 12h424c6.6 0 12-5.4 12-12z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                </FontAwesomeIcon>
                <div
                  className="updated-at"
                >
                  Updated: 12-12-2018
                </div>
              </span>
            </div>
            <div
              className="article-taglist"
            >
              <span
                key="tag"
              >
                tag
              </span>
            </div>
            <div
              className="article-body text-justify"
            >
              <div
                className="dots text-center large"
              >
                 ... 
              </div>
              body
            </div>
            <div
              className="article-stats"
            >
              <LikeDislike
                dislikes={2}
                likes={12}
                onLikeDislike={[Function]}
              >
                <div
                  className="article-like-dislike"
                >
                  <form>
                    <button
                      className="thumbs mr-2"
                      id="bt-like"
                      onClick={[Function]}
                      type="button"
                    >
                      <FontAwesomeIcon
                        border={false}
                        className=""
                        fixedWidth={false}
                        flip={null}
                        icon="thumbs-up"
                        inverse={false}
                        listItem={false}
                        mask={null}
                        pull={null}
                        pulse={false}
                        rotation={null}
                        size={null}
                        spin={false}
                        symbol={false}
                        title=""
                        transform={null}
                      >
                        <svg
                          aria-hidden="true"
                          className="svg-inline--fa fa-thumbs-up fa-w-16 "
                          data-icon="thumbs-up"
                          data-prefix="fas"
                          role="img"
                          style={Object {}}
                          viewBox="0 0 512 512"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M104 224H24c-13.255 0-24 10.745-24 24v240c0 13.255 10.745 24 24 24h80c13.255 0 24-10.745 24-24V248c0-13.255-10.745-24-24-24zM64 472c-13.255 0-24-10.745-24-24s10.745-24 24-24 24 10.745 24 24-10.745 24-24 24zM384 81.452c0 42.416-25.97 66.208-33.277 94.548h101.723c33.397 0 59.397 27.746 59.553 58.098.084 17.938-7.546 37.249-19.439 49.197l-.11.11c9.836 23.337 8.237 56.037-9.308 79.469 8.681 25.895-.069 57.704-16.382 74.757 4.298 17.598 2.244 32.575-6.148 44.632C440.202 511.587 389.616 512 346.839 512l-2.845-.001c-48.287-.017-87.806-17.598-119.56-31.725-15.957-7.099-36.821-15.887-52.651-16.178-6.54-.12-11.783-5.457-11.783-11.998v-213.77c0-3.2 1.282-6.271 3.558-8.521 39.614-39.144 56.648-80.587 89.117-113.111 14.804-14.832 20.188-37.236 25.393-58.902C282.515 39.293 291.817 0 312 0c24 0 72 8 72 81.452z"
                            fill="currentColor"
                            style={Object {}}
                          />
                        </svg>
                      </FontAwesomeIcon>
                    </button>
                    <span
                      className="mr-4"
                    >
                      12
                    </span>
                    <button
                      className="thumbs mr-2"
                      id="bt-dislike"
                      onClick={[Function]}
                      type="button"
                    >
                      <FontAwesomeIcon
                        border={false}
                        className=""
                        fixedWidth={false}
                        flip={null}
                        icon="thumbs-down"
                        inverse={false}
                        listItem={false}
                        mask={null}
                        pull={null}
                        pulse={false}
                        rotation={null}
                        size={null}
                        spin={false}
                        symbol={false}
                        title=""
                        transform={null}
                      >
                        <svg
                          aria-hidden="true"
                          className="svg-inline--fa fa-thumbs-down fa-w-16 "
                          data-icon="thumbs-down"
                          data-prefix="fas"
                          role="img"
                          style={Object {}}
                          viewBox="0 0 512 512"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M0 56v240c0 13.255 10.745 24 24 24h80c13.255 0 24-10.745 24-24V56c0-13.255-10.745-24-24-24H24C10.745 32 0 42.745 0 56zm40 200c0-13.255 10.745-24 24-24s24 10.745 24 24-10.745 24-24 24-24-10.745-24-24zm272 256c-20.183 0-29.485-39.293-33.931-57.795-5.206-21.666-10.589-44.07-25.393-58.902-32.469-32.524-49.503-73.967-89.117-113.111a11.98 11.98 0 0 1-3.558-8.521V59.901c0-6.541 5.243-11.878 11.783-11.998 15.831-.29 36.694-9.079 52.651-16.178C256.189 17.598 295.709.017 343.995 0h2.844c42.777 0 93.363.413 113.774 29.737 8.392 12.057 10.446 27.034 6.148 44.632 16.312 17.053 25.063 48.863 16.382 74.757 17.544 23.432 19.143 56.132 9.308 79.469l.11.11c11.893 11.949 19.523 31.259 19.439 49.197-.156 30.352-26.157 58.098-59.553 58.098H350.723C358.03 364.34 384 388.132 384 430.548 384 504 336 512 312 512z"
                            fill="currentColor"
                            style={Object {}}
                          />
                        </svg>
                      </FontAwesomeIcon>
                    </button>
                    <span
                      className=""
                    >
                      2
                    </span>
                  </form>
                </div>
              </LikeDislike>
              <Connect(DeleteArticlePage)
                articleId={1}
              >
                <DeleteArticlePage
                  articleDelete={[Function]}
                  articleId={1}
                  deleteArticleReducer={Object {}}
                >
                  <div>
                    <DeleteArticle
                      onClick={[Function]}
                    >
                      <div>
                        <button
                          className="btn btn-danger"
                          onClick={[Function]}
                          type="submit"
                        >
                          DELETE ARTICLE
                        </button>
                      </div>
                    </DeleteArticle>
                  </div>
                </DeleteArticlePage>
              </Connect(DeleteArticlePage)>
              <div
                className="article-views"
              >
                <FontAwesomeIcon
                  border={false}
                  className=""
                  fixedWidth={false}
                  flip={null}
                  icon="eye"
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-eye fa-w-18 "
                    data-icon="eye"
                    data-prefix="fas"
                    role="img"
                    style={Object {}}
                    viewBox="0 0 576 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M569.354 231.631C512.969 135.949 407.81 72 288 72 168.14 72 63.004 135.994 6.646 231.631a47.999 47.999 0 0 0 0 48.739C63.031 376.051 168.19 440 288 440c119.86 0 224.996-63.994 281.354-159.631a47.997 47.997 0 0 0 0-48.738zM288 392c-75.162 0-136-60.827-136-136 0-75.162 60.826-136 136-136 75.162 0 136 60.826 136 136 0 75.162-60.826 136-136 136zm104-136c0 57.438-46.562 104-104 104s-104-46.562-104-104c0-17.708 4.431-34.379 12.236-48.973l-.001.032c0 23.651 19.173 42.823 42.824 42.823s42.824-19.173 42.824-42.823c0-23.651-19.173-42.824-42.824-42.824l-.032.001C253.621 156.431 270.292 152 288 152c57.438 0 104 46.562 104 104z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                </FontAwesomeIcon>
                <span
                  className="pl-2"
                >
                  # Views
                </span>
              </div>
            </div>
          </div>
        </div>
      </Article>
    </ArticlePage>
  </Connect(ArticlePage)>
</Provider>
`;
